test/acl_mem_test.cpp

```diff
@@ -2799,6 +2799,57 @@ TEST(acl_mem, case_205751_overlapping_alloc) {
   CHECK_EQUAL(CL_SUCCESS, clReleaseMemObject(c));
 }
 
+TEST(acl_mem, buffer_location_property) {
+  ACL_LOCKED(acl_print_debug_msg("begin buffer_location_property\n"));
+  // Test assumes more than 1 global memory space
+  // Allocate a small buffer (a), then try to allocate two buffers (b, c) of
+  // size bank_size.  Expect the second allocation to fail.
+  cl_mem a;
+  cl_int status = CL_SUCCESS;
+  size_t total_size = ACL_RANGE_SIZE(
+      m_device[0]->def.autodiscovery_def.global_mem_defs[0].range);
+  size_t bank_size = total_size / 2;
+  size_t small_size = bank_size / 1024;
+
+  cl_mem_properties_intel props[] = {CL_MEM_ALLOC_BUFFER_LOCATION_INTEL, 0, 0};
+  a = clCreateBufferWithPropertiesINTEL(m_context, props, 0, bank_size, 0,
+                                        &status);
+  ACL_LOCKED(CHECK(acl_mem_is_valid(a)));
+  CHECK_EQUAL(CL_SUCCESS, status);
+  if (a) {
```

Comment:
This would the test if for some reason `a` is `NULL`.

You can use `assert(a)` here and below to quiesce Klocwork.

Reply:
Thank you! (sorry I totally forgot the convention about just using assert in test files). Now the changes are ready, and checks passed.